File: apps/core-api/src/Akademik.DataProvider/Models/Assignment.cs
namespace Akademik.DataProvider.Models;

public sealed class Assignment
{
	public int Id { get; set; }
	public required int UserId { get; set; }
	public required int RoomId { get; set; }
	public DateOnly StartDate { get; set; }
	public DateOnly? EndDate { get; set; }
	public bool IsActive { get; set; }
	public User? User { get; set; }
	public Room? Room { get; set; }
}

File: apps/web/src/shared/types/assignment.ts
import { Room } from "./room"
import { User } from "./user"

export interface PaginationParams {
    pageNumber: number;
    pageSize: number;
}

export interface PagedResult<T> {
    items: T[];
    count: number;
}

export interface Assignment {
    id: number;
    userId: number;
    roomId: number;
    startDate: string;
    endDate?: string | null;
    isActive: boolean;
    user: User;
    room: Room;
}

export interface CreateAssignmentRequest {
    userId: number;
    roomId: number;
    startDate: string | Date;
}